events/schema.md

```diff
@@ -4,30 +4,157 @@ Events published to the bus have a standard schema that allows SDKs and services
+            "identifier": "10627",
+        },
+        {
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": "1",
+        },
+        {
+            "service": "article-store",
+            "name": "article-run",
+            "identifier": "6551f09a-bd23-4f12-a039-f8fa78ef776a",
+        },
+        {
+            "service": "search",
+            "name": "indexing",
+            "identifier": "18b21fea-5a87-11e8-9c2d-fa7ae01bbebc",
+        },
+    ],
+}
+```
+
+### Example: downstream deposit service pushing an article to Crossref
+
+```
+{
+    "eventId": "448278c4-22d6-11e8-b467-0ed5f89f718b",
+    "happenedAt": "2018-03-08T12:00:00+00:00",
+    "aggregate": {
+        "service": "downstream-sample",
+        "name": "deposit",
+        "identifier": "2018-03-08/1", # daily attempts
+    },
+    "type": "downstream-crossref-started",
+    "data": {
+        ... 
+    },
+    "correlationIds": [
+        {
+            "service": "article-store",
+            "name": "article",
+            "identifier": "10627",
+        },
+        {
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": "1",
+        },
+        {
+            "service": "article-store",
+            "name": "article-run",
+            "identifier": "6551f09a-bd23-4f12-a039-f8fa78ef776a",
+        },
+        {
+            "service": "downstream-sample",
+            "name": "deposit",
+            "identifier": "2018-03-08/1",
+        },
+    ],
+}
+```
+
+## Fields
+
+### `eventId`
+
+Unique for all events. It is a UUID version 1 generated with time and a node identifier.
+
+### `happenedAt` 
+
+Identifies when the event has happened (then it can get published, consumed, etc). It follows the `yyyy-mm-ddTHH:MM:SS+00:00` ISO 8601 format, and is expressed in UTC.
+
+### `aggregate`
+
+Identifies the unit of consistency that has produced the event: article run in the bot, article version in the articles store, podcast episode in journal-cms, an article's set of metrics in metrics, a single profile in profiles.
+
+All three information are required to uniquely identify it:
+
+- `service`: `[a-z\-]+`
+- `name`: `[a-z\-]+`
+- `identifier`: `.+` (possibly URL-friendly)
+
+### `type`
+
+Describes what the event is about so that they can be grouped or recognized. It follows the `[a-z\-\.]+` regular expression, with `.` separating hierarchical levels and `-` separating words in a single level (if there are multiple words).
+
+### `data` (optional)
+
+Is opaque here, and can contain anything as long as it follows naming conventions. Events with the same `type` should usually follow the same schema.
+
+### `correlationIds` (optional)
+
+Can be used to track events across services, grouping them by the kind of session that originated them or the data type they refer to. Could be an article run during ingestion, or a journal page creating a profile.
```

Comment:
Might be able to solve https://github.com/libero/walking-skeleton/issues/31#issuecomment-390585994 with a consistent `started`/`completed`/`failed` naming in `type` and `aggregate`? So the aggregate is the process, so an 'ingest article' aggregate has a sub-process of 'deposit images', both of these having `started`/`completed` events, and the latter also has the correlation ID of the former.

The dashboard can then construct something like:

- [article.ingest aggregate]
    - article.ingest.start
        - [article.ingest.image aggregate]
            - article.ingest.image.start
            - article.ingest.image.completed
    - article.ingest.completed